src/operator/batch_norm.cu

```diff
@@ -117,6 +117,11 @@ struct GradOp {
   const DeviceTensor gradOutput;
 };
 
+#if CUDA_VERSION >= 9000
+#define FULLMASK 0xFFFFFFFF
+#define __shfl_xor(...) __shfl_xor_sync(FULLMASK, __VA_ARGS__)
```

Comment:
I would not want to do that - somebody might add __shfl_xor later to the code with no knowledge about the cooperative groups. Normally this will result in compiler warning about unsafe code, but if we make this code global, then it will compile without warnings, and may lead to deadlocks when used.